runtime/compiler/runtime/JITServerAOTDeserializer.cpp

```diff
@@ -853,24 +877,28 @@ JITServerAOTDeserializer::updateSCCOffsets(SerializedAOTMethod *method, TR::Comp
       {
       // Get the SCC offset of the corresponding entity in the local SCC
       const SerializedSCCOffset &serializedOffset = method->offsets()[i];
-      uintptr_t sccOffset = getSCCOffset(serializedOffset.recordType(), serializedOffset.recordId(), wasReset);
-      if (sccOffset == (uintptr_t)-1)
-         return false;
-
-      // Update the SCC offset stored in AOT method relocation data
-      uint8_t *ptr = start + serializedOffset.reloDataOffset();
-      TR_ASSERT_FATAL((ptr >= start + sizeof(uintptr_t)/*skip the size word*/) && (ptr < end),
-                      "Out-of-bounds relocation data offset %zu in serialized method %s",
-                      serializedOffset.reloDataOffset(), comp->signature());
+      // Thunks do not use their SCC offset entries
+      if (serializedOffset.recordType() != Thunk)
```

Comment:
This could be rewritten as `if (serializedOffset.recordType() == AOTSerializationRecordType::Thunk) continue;` to avoid shifting the code below by an extra level of indentation. Note the `AOTSerializationRecordType::` (to match existing code).